modules/apps/social/social-bookmarks-taglib/src/main/java/com/liferay/social/bookmarks/taglib/internal/util/SocialBookmarksRegistryImpl.java

```diff
@@ -55,66 +51,26 @@ public SocialBookmark getSocialBookmark(String type) {
 		return socialBookmark;
 	}
 
-	@Override
-	public List<SocialBookmark> getSocialBookmarks() {
-		List<SocialBookmark> socialBookmarks = new ArrayList<>();
-
-		for (String type : getSocialBookmarksTypes()) {
-			socialBookmarks.add(getSocialBookmark(type));
-		}
-
-		return socialBookmarks;
-	}
-
 	@Override
 	public List<String> getSocialBookmarksTypes() {
-		Set<String> socialBookmarksTypes = new LinkedHashSet<>();
-
-		for (String type : _serviceTrackerList) {
-			socialBookmarksTypes.add(type);
-		}
-
-		for (String type : PropsUtil.getArray(_SOCIAL_BOOKMARK_TYPES)) {
-			if (_isValidDeprecatedSocialBookmark(type)) {
-				socialBookmarksTypes.add(type);
-			}
-		}
-
-		return new ArrayList<>(socialBookmarksTypes);
+		return _serviceTrackerList.toList();
 	}
 
 	@Activate
 	protected void activate(BundleContext bundleContext) {
 		_serviceTrackerList = ServiceTrackerListFactory.open(
-			bundleContext, SocialBookmark.class, null,
-			new SocialBookmarkTypeServiceTrackerCustomizer(),
+			bundleContext, SocialBookmark.class, "(social.bookmarks.type=*)",
```

Comment:
@W2jx Why not keep using a ServiceTrackerMap as you are just using this service property as the key?

Reply:
Hi @dantewang , Because by the original logic, we want the list of "social.bookmarks.type" to be sorted by "social.bookmarks.priority", but there's no way for ServiceTrackerMap to sort its key